src/docfx/lib/HttpClientUtility.cs

```diff
@@ -11,49 +11,36 @@ namespace Microsoft.Docs.Build
 {
     internal static class HttpClientUtility
     {
-        private const int RetryCount = 3;
-        private const int RetryInterval = 1000;
         private static readonly HttpClient s_httpClient = new HttpClient();
 
-        public static async Task<HttpResponseMessage> GetAsync(string requestUri, Config config, string etag = null)
+        public static async Task<HttpResponseMessage> GetAsync(string requestUri, Config config, EntityTagHeaderValue etag = null)
         {
-            HttpResponseMessage response;
-            for (var i = 0; i < RetryCount; i++)
-            {
-                try
+            return await RetryUtility.Retry(
+                () =>
                 {
                     // Create new instance of HttpRequestMessage to avoid System.InvalidOperationException:
                     // "The request message was already sent. Cannot send the same request message multiple times."
                     var message = CreateHttpRequestMessage(requestUri, config);
                     message.Method = HttpMethod.Get;
-                    if (!string.IsNullOrEmpty(etag))
+                    if (etag != null)
                     {
-                        message.Headers.IfNoneMatch.Add(new EntityTagHeaderValue(etag));
+                        message.Headers.IfNoneMatch.Add(etag);
                     }
-                    response = await s_httpClient.SendAsync(message);
-                }
-                catch (Exception ex) when (
-                    i < RetryCount - 1 &&
-                    (ex is HttpRequestException ||
-                     ex is TimeoutException ||
-                     ex is OperationCanceledException))
-                {
-                    await Task.Delay(RetryInterval);
-                    continue;
-                }
-                return response;
-            }
-
-            Debug.Fail("should never reach here");
-            return null;
+                    return s_httpClient.SendAsync(message);
+                },
+                IsExceptionsToRetry);
```

Comment:
Put it inline here since it is not long?  Also, exceptions are plural, naming is wrong